db/db_impl/db_impl_write.cc

```
@@ -1954,12 +1962,65 @@ void DBImpl::NotifyOnMemTableSealed(ColumnFamilyData* /*cfd*/,
+  memtable_info.num_deletes = cfd->mem()->num_deletes();
+#endif  // ROCKSDB_LITE
+  int num_imm_unflushed = cfd->imm()->NumNotFlushed();
+  SequenceNumber seq = versions_->LastSequence();
+  new_mem = cfd->ConstructNewMemtable(mutable_cf_options, seq);
+  assert(new_mem != nullptr);
+  context->superversion_context.NewSuperVersion();
+  ROCKS_LOG_INFO(immutable_db_options_.info_log,
+                 "[%s] New memtable created with log file: #%" PRIu64
+                 ". Immutable memtables: %d.\n",
+                 cfd->GetName().c_str(), next_log_num, num_imm_unflushed);
+  log_write_mutex_.Lock();
+  logfile_number_ = next_log_num;
+  // alive_log_files_ and logs_ size should be 1, so updating front is exactly the same as
+  // updating back
+  alive_log_files_.front().number = next_log_num;
+  logs_.front().number = next_log_num;
+  log_write_mutex_.Unlock();
+
+  cfd->mem()->SetNextLogNumber(logfile_number_);
+  cfd->mem()->SetReplicationSequence(replication_sequence);
+  cfd->imm()->Add(cfd->mem(), &context->memtables_to_free_);
+  new_mem->Ref();
+  cfd->SetMemtable(new_mem);
+  InstallSuperVersionAndScheduleWork(cfd, &context->superversion_context,
+                                     mutable_cf_options);
+#ifndef ROCKSDB_LITE
+  mutex_.Unlock();
+  // Notify client that memtable is sealed, now that we have successfully
+  // installed a new memtable
+  NotifyOnMemTableSealed(cfd, memtable_info);
+  mutex_.Lock();
+#endif  // ROCKSDB_LITE
+  return Status::OK();
+}
+
 // REQUIRES: mutex_ is held
 // REQUIRES: this thread is currently at the front of the writer queue
 // REQUIRES: this thread is currently at the front of the 2nd writer queue if
 // two_write_queues_ is true (This is to simplify the reasoning.)
-Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context,
-                              std::string replication_sequence) {
+Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context) {
```

Could you return an error from `SwitchMemtable` if `replication_log_listener` is set?